idea/src/org/jetbrains/kotlin/idea/intentions/ConvertArrayParameterToVarargIntention.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.builtins.KotlinBuiltIns
+import org.jetbrains.kotlin.builtins.PrimitiveType
+import org.jetbrains.kotlin.idea.search.usagesSearch.descriptor
+import org.jetbrains.kotlin.js.descriptorUtils.nameIfStandardType
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.getChildOfType
+
+class ConvertArrayParameterToVarargIntention : SelfTargetingIntention<KtParameter>(
+    KtParameter::class.java, "Convert to vararg parameter"
+) {
+
+    override fun isApplicableTo(element: KtParameter, caretOffset: Int): Boolean {
+        val typeReference = element.getChildOfType<KtTypeReference>() ?: return false
+        if (element.parent.parent is KtFunctionLiteral) return false
+        if (element.isVarArg) return false
+
+        val type = element.descriptor?.type ?: return false
+        return when {
+            KotlinBuiltIns.isPrimitiveArray(type) ->
+                true
+            KotlinBuiltIns.isArray(type) -> {
+                val typeArguments = typeReference.typeElement?.typeArgumentsAsTypes
+                val typeProjection = typeArguments?.firstOrNull()?.parent as? KtTypeProjection
+                typeProjection?.hasModifier(KtTokens.IN_KEYWORD) == false
```

Comment:
You are right about `in`, but we can also have problems when no variance is stated. E.g.

```
fun bar(arr: Array<String>) {
    arr[0] = "" // arr cannot be made vararg here
}
```

With `out` variance, this assignment is not possible, but with no variance, we have to check function body for a possible assignment to array elements :(

Reply:
I think that for performance reasons it's not a good idea to implement such a check. So I suggest to change intention message in case no variance is stated, adding something like `(may break code)` at the end.